src/sage/geometry/hyperplane_arrangement/arrangement.py

```diff
@@ -1281,8 +1334,18 @@ def restriction(self, hyperplane):
             hyperplanes.append([A, b])
         names = list(parent._names)
         names.pop(pivot)
-        H = HyperplaneArrangements(parent.base_ring(), names=tuple(names))
-        return H(*hyperplanes, signed=False, backend=self._backend)
+        if 'Ordered' in str(type(self)):
```

Comment:
No reason but my attempts to use `isinstance` fail.

Reply:
You are doing something wrong then. What have you tried? (This is a somewhat brittle check since if I use a subclass called `OrderedFoo`, it will go into this case.)

Reply:
I found the problem, done